torch/_inductor/codegen/cpp_gemm_template.py

```diff
@@ -395,11 +406,26 @@ def render(  # type: ignore[override]
             Y = template_buffer_node
 
         template_buffer = Y
+        gemm_output_buffer = template_buffer
+
+        epilogues: List[ir.IRNode] = []
+        if self.epilogue_creator is not None:
+            gemm_output_name = "GemmOut"
```

Comment:
with in-template epilogue nodes, the gemm output could be different from the template output, i.e., gemm out -> in-template epilogues -> template output -> fused out-of-template epilogues. So we create a dedicated buffer for gemm out.